### ISSUE TITLE
project 2

### ISSUE THREAD

**Issue by Legend700** [Legend700/Datacharlotte#3]
Project 2: Determining the Best Sport Through Data Analysis

1. Introduction (10 pts)

Problem Statement:
The world of sports is diverse, with each sport requiring different levels of physical fitness, endurance, and skill. But what makes a sport the “best”? Is it popularity, physical demands, or career longevity?
Objective:
This project aims to analyze various sports using data-driven methods to determine which sport stands out based on performance metrics, injury risk, training requirements, and popularity.
Key Questions:
Which sport requires the highest endurance, speed, or strength?
Which sport has the lowest injury risk while maintaining high performance?
Does training intensity correlate with better performance?
Which sport offers the best balance between career longevity and financial success?
2. Dataset Introduction (10 pts)

Source of Data:
The dataset used for this project was created to include key sports-related metrics such as speed, endurance, training hours, and popularity.
Download link: Sports Dataset
Dataset Overview:
The dataset consists of the following features:
Sport: Name of the sport
Avg Speed (m/s): Average movement speed of an athlete in the sport
Endurance (1-10): A rating of how much stamina the sport requires
Strength (kg): Estimated average strength requirement in kilograms
Training Hours/Week: Average hours per week athletes train
Injury Risk (%): Percentage likelihood of injury per season
Global Popularity (1-100): Popularity rating based on global fanbase and participation
Avg Salary ($): The average salary of professional athletes in the sport
Career Longevity (Years): Estimated number of years an athlete can compete professionally
3. Pre-processing the Data (15 pts)

Handling Missing Values:
Checked for missing values and filled them using median values to prevent data loss.
Standardization:
Scaled numerical features like strength, speed, and salary to bring them into comparable ranges.
Encoding Categorical Data:
Since "Sport" is a categorical variable, converted it to numerical values for machine learning models.
Outlier Detection:
Used box plots to detect extreme values in salary and training hours that could skew results.
4. Data Understanding & Visualization (15 pts)

Exploratory Data Analysis (EDA):
Correlation Matrix: Checked relationships between training hours, injury risk, and performance metrics.
Histograms & Boxplots: Visualized distributions of salary, career longevity, and injury risk.
Scatter Plots: Examined relationships between training hours and performance.
Key Findings:
Track & Field had the highest speed but lowest popularity.
American Football had the highest injury rate (30%) due to high contact.
Soccer was one of the most well-balanced sports in terms of endurance, popularity, and salary.
5. Storytelling (20 pts)

What Have We Learned?
Some of the most physically demanding sports (e.g., Swimming, Track & Field) are not the most financially rewarding.
Popularity does not always align with the most physically demanding sports.
High training hours do not always lead to high salaries.
Answering Initial Questions:
The most physically demanding sports based on speed, endurance, and strength: Track & Field, Soccer, Swimming
The sport with the lowest injury risk while maintaining performance: Swimming
The best financial reward and longevity balance: Basketball
The best all-around sport (considering physicality, injury risk, and popularity): Soccer
6. Impact Section (5 pts)

Potential Impact:
These insights could help young athletes make informed career choices.
The study highlights safety concerns in sports with high injury risks.
Possible Biases & Missing Data:
The dataset does not account for psychological stress in different sports.
Some niche sports with dedicated followings were not included.
7. References (5 pts)

Data generated based on estimated values from sports analytics reports.
Further research can be conducted using public datasets from FIFA, NBA, and Olympic records.
8. Code & Notebook Submission (10 pts)

Download Dataset: Sports Dataset
Code snippets will be included in the final report submission.

[sports_dataset.csv](https://github.com/user-attachments/files/18967346/sports_dataset.csv)

# Importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, mean_squared_error

# Loading the dataset
file_path = "/mnt/data/sports_dataset.csv"  # Replace with your actual dataset path
df = pd.read_csv(file_path)

# Checking for missing values and basic info
print(df.isnull().sum())  # Check for missing values
print(df.info())  # Data types and other info
print(df.describe())  # Statistical summary

# Pre-processing the data
# Fill missing values with median for numerical columns
df.fillna(df.median(), inplace=True)

# Standardizing numerical features (Scaling the data)
scaler = StandardScaler()
numerical_cols = ['Avg Speed (m/s)', 'Strength (kg)', 'Training Hours/Week', 
                  'Injury Risk (%)', 'Global Popularity (1-100)', 
                  'Avg Salary ($)', 'Career Longevity (Years)']
df[numerical_cols] = scaler.fit_transform(df[numerical_cols])

# Encoding the 'Sport' column as a categorical variable (One-Hot Encoding)
df_encoded = pd.get_dummies(df, columns=['Sport'], drop_first=True)

# Exploratory Data Analysis (EDA)

# Visualizing the correlation matrix
plt.figure(figsize=(10, 6))
sns.heatmap(df_encoded.corr(), annot=True, cmap='coolwarm', fmt='.2f')
plt.title('Correlation Matrix')
plt.show()

# Visualizing distribution of Injury Risk across sports
plt.figure(figsize=(10, 6))
sns.boxplot(x='Sport', y='Injury Risk (%)', data=df)
plt.title('Injury Risk Distribution by Sport')
plt.xticks(rotation=45)
plt.show()

# Scatter plot to visualize the relationship between Training Hours and Performance (Avg Speed)
plt.figure(figsize=(10, 6))
sns.scatterplot(x='Training Hours/Week', y='Avg Speed (m/s)', data=df)
plt.title('Training Hours vs Speed')
plt.show()

# Analyzing the impact of training hours on salary
plt.figure(figsize=(10, 6))
sns.scatterplot(x='Training Hours/Week', y='Avg Salary ($)', data=df)
plt.title('Training Hours vs Salary')
plt.show()

# Model Plan (e.g., Classification: Predicting the Sport with highest injury risk)
# Splitting data into features (X) and target (y)
X = df_encoded.drop(['Injury Risk (%)'], axis=1)  # Feature set
y = df_encoded['Injury Risk (%)']  # Target variable

# Splitting into training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Model: Logistic Regression (For classification purposes)
model = LogisticRegression()
model.fit(X_train, y_train)

# Predicting on test set
y_pred = model.predict(X_test)

# Evaluating the model
accuracy = accuracy_score(y_test, y_pred)
print(f'Accuracy of the Logistic Regression model: {accuracy * 100:.2f}%')

# Mean Squared Error for Regression tasks (if applicable)
y_pred_continuous = model.predict(X_test)  # Regression can be used for continuous targets
mse = mean_squared_error(y_test, y_pred_continuous)
print(f'Mean Squared Error (MSE): {mse:.2f}')

# Storytelling: Key insights
# Top 3 sports with lowest injury risk
top_sports = df[['Sport', 'Injury Risk (%)']].groupby('Sport').mean().sort_values(by='Injury Risk (%)').head(3)
print('Top 3 sports with lowest injury risk:')
print(top_sports)

# Impact: Analyzing the possible harm and missing data
# Discuss any potential data biases or shortcomings
print("\nPossible Data Biases and Shortcomings:")
print("This dataset does not consider psychological stress or data about amateur players. High injury risk could be due to the nature of professional leagues, and some niche sports may not be included.")
